test/integration/upgrading.rb

```diff
@@ -35,9 +35,28 @@ class UpgradeTests < EinhornIntegrationTestCase
       @port = find_free_port
       @server_program = File.join(@dir, "env_printer.rb")
       @socket_path = File.join(@dir, "einhorn.sock")
+
+      mangler = ('a'..'z').to_a.shuffle[0,8].join
+      @unix_listener_socket_path = "unix-listener-einhorn-#{mangler}.sock"
```

Comment:
this is in `pwd` rather than `@dir` because of unix domain socket length limitations.
I considered the abstract socket namespace, but didn't go with it, since I figured this was a more common pattern.